app/BUILD.bazel

```diff
@@ -336,6 +388,70 @@ kt_android_library(
     visibility = ["//visibility:private"],
 )
 
+# Library for all custom data-binding adapters under org/oppia/app/databinding
+android_library(
+    name = "databinding_adapters",
+    custom_package = "org.oppia.app.impl.adapters",
+    srcs = DATABINDING_ADAPTER_FILES,
+    enable_data_binding = 1,
+    manifest = "src/main/DbAdaptersAndroidManifest.xml",
+    deps = [
+        ":dagger",
+        "//model",
+        "//utility",
+        ":view_models",
+        ":resources",
+        artifact("androidx.appcompat:appcompat"),
+        artifact("com.chaos.view:pinview:1.4.3"),
+        artifact("com.github.bumptech.glide:glide"),
+        artifact("com.caverock:androidsvg-aar"),
+        artifact("androidx.lifecycle:lifecycle-livedata-ktx"),
+        artifact("androidx.databinding:databinding-runtime"),
+        artifact("androidx.databinding:databinding-adapters"),
+        artifact("androidx.databinding:databinding-common"),
+        artifact("androidx.annotation:annotation"),
+        artifact("com.google.android.material:material:1.2.0-alpha02"),
+        artifact("androidx.constraintlayout:constraintlayout:1.1.3"),
+        artifact("de.hdodenhof:circleimageview:3.0.1"),
+        artifact("javax.annotation:javax.annotation-api:jar"),
+        artifact("org.jetbrains:annotations:jar"),
+        "@circularimageview//circularimageview:circular_image_view",
+    ],
+    visibility = ["//visibility:public"],
```

Comment:
Here & elsewhere: does any package in app actually need to be public other than the top-level library included in the ``android_binary`` target?

Reply:
Fixed! They can all be private except for `app`. :)